code/modules/power/apc.dm

```diff
@@ -961,6 +961,15 @@ var/global/list/all_apcs = list()
 			environ = state
 	force_update_channels()
 
-
+/obj/machinery/power/apc/area_changed()
+	. = ..()
+	//Make sure to resume processing if our area changed to something else than null
+	if(QDELETED(src))
+		return
+	if(area)
+		if(is_processing && !(processing_flags & MACHINERY_PROCESS_SELF))
+			processing_flags |= MACHINERY_PROCESS_SELF //don't use START_PROCESSING_MACHINE here or it'll print a stack trace for being already in the processing list
```

Comment:
Should only runtime if it's processing on a different SS, which isn't relevant here.

Reply:
You're right, I misread the macro -_- Pushing a fix in a bit.